File: Sample-ObjC/05-query-user/Mesosfer-ObjC/LoginViewController.m
/**
 * Copyright (c) 2016, Mesosfer
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "LoginViewController.h"
#import "UIViewController+Helper.h"
#import <Mesosfer/Mesosfer.h>

@implementation LoginViewController

- (IBAction)handleLoginButton:(UIButton *)sender {
    NSString *email = self.textEmailAddress.text;
    NSString *password = self.textPassword.text;
    
    [MFUser logInAsyncWithEmail:email password:password block:^(MFUser * _Nullable user, NSError * _Nullable error) {
        if (error) {
            [self showErrorWithTitle:@"Error login" error:error handler:nil];
            return;
        }
        
        [self dismissViewControllerAnimated:YES completion:nil];
    }];
}
@end

File: Sample-Swift/04-fetch-n-update-user/Mesosfer-Swift/LoginViewController.swift
/**
 * Copyright (c) 2016, Mesosfer
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 */

import Foundation
import UIKit
import Mesosfer

class LoginViewController: UIViewController {
    @IBOutlet weak var textEmailAddress: UITextField!
    @IBOutlet weak var textPassword: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func handleLoginButton(_ sender: UIButton) {
        let email = self.textEmailAddress.text
        let password = self.textPassword.text
        
        MFUser.logInAsync(withEmail: email!, password: password!) { (user, error) in
            if let e = error as? NSError {
                self.showError(title: "Error login", error: e)
                return
            }
            
            // dismiss this controller
            self.dismiss(animated: true)
        }
    }
}
